internal/cloudapi/v2/handler.go

```diff
@@ -1178,3 +1186,183 @@ func genRepoConfig(repo Repository) (*rpmmd.RepoConfig, error) {
+	if err != nil {
+		return HTTPErrorWithInternal(ErrorGettingOSBuildJobStatus, err)
+	}
+
+	if osbuildStatus.Finished.IsZero() || !osbuildResult.Success {
+		return HTTPError(ErrorComposeBadState)
+	}
+
+	us, err := uploadStatusFromOSBuildJobResult(&osbuildResult)
+	if err != nil {
+		return err
+	}
+
+	// the id of the last job in the dependency chain which users should wait on
+	finalJob := jobId
+	// look at the upload status of the osbuild dependency to decide what to do
+	switch us.Type {
+	case UploadTypesAws:
+		options := us.Options.(AWSEC2UploadStatus)
+		var img AWSEC2ImageFromCompose
+		err := ctx.Bind(&img)
+		if err != nil {
+			return err
+		}
+
+		shareAmi := options.Ami
+		shareRegion := img.Region
+		if img.Region != options.Region {
+			// Let the share job use dynArgs
+			shareAmi = ""
+			shareRegion = ""
+
+			// Check deps if we need to do a copyjob
+			foundDep := false
+			for _, d := range deps {
+				jt, err := h.server.workers.JobType(d)
+				if err != nil {
+					return HTTPErrorWithInternal(ErrorGettingJobType, err)
+				}
+				if jt == worker.JobTypeAWSEC2Copy {
+					var cjResult worker.AWSEC2CopyJobResult
+					_, _, err := h.server.workers.AWSEC2CopyJobStatus(d, &cjResult)
+					if err != nil {
+						return HTTPErrorWithInternal(ErrorGettingAWSEC2JobStatus, err)
+					}
+
+					if cjResult.JobError == nil && options.Region == cjResult.Region {
+						finalJob = d
+						foundDep = true
+						break
+					}
+				}
+			}
+
+			if !foundDep {
+				copyJob := &worker.AWSEC2CopyJob{
+					Ami:          options.Ami,
+					SourceRegion: options.Region,
+					TargetRegion: img.Region,
+					TargetName:   fmt.Sprintf("composer-api-%s", uuid.New().String()),
+				}
+				finalJob, err = h.server.workers.EnqueueAWSEC2CopyJob(copyJob, finalJob, channel)
+				if err != nil {
+					return HTTPErrorWithInternal(ErrorEnqueueingJob, err)
+				}
+			}
+		}
+
+		if img.ShareWithAccounts != nil && len(*img.ShareWithAccounts) > 0 {
+			shareJob := &worker.AWSEC2ShareJob{
+				Ami:               shareAmi,
+				Region:            shareRegion,
+				ShareWithAccounts: *img.ShareWithAccounts,
+			}
+			finalJob, err = h.server.workers.EnqueueAWSEC2ShareJob(shareJob, finalJob, channel)
+			if err != nil {
+				return HTTPErrorWithInternal(ErrorEnqueueingJob, err)
+			}
+		}
+	default:
+		return HTTPError(ErrorUnsupportedImage)
+	}
+
+	return ctx.JSON(http.StatusCreated, ImageFromComposeResponse{
+		ObjectReference: ObjectReference{
+			Href: fmt.Sprintf("/api/image-builder-composer/v2/composes/%v/image", jobId),
+			Id:   finalJob.String(),
+			Kind: "ImageFromComposeId",
+		},
+		Id: finalJob.String(),
+	})
+}
+
+func (h *apiHandlers) GetImageFromComposeStatus(ctx echo.Context, id string) error {
+	return h.server.EnsureJobChannel(h.getImageFromComposeStatus)(ctx, id)
+}
+
+func (h *apiHandlers) getImageFromComposeStatus(ctx echo.Context, id string) error {
```

Comment:
Is there a reason why we can't or don't want to treat the ID returned by the `compose/{id}/image` endpoint as another compose ID?

If we did, like we do for Koji composes, we could get rid of the new status endpoint and just extend the `/compose/{id}` endpoint to handle new job type, in addition to the osbuild job and koji-finalize job.

Reply:
Makes sense, no  need to duplicate this everywhere.

Reply:
hm, going through it, it does return a different kind of object, just the upload status part really.

Reply:
`compose/{id}/image` makes sense to me, as an image always belongs to an existing compose. Also, the returned data are different, so I would rather not merge them.